### ISSUE TITLE
Improving css, form.js

### ISSUE THREAD

**Issue by ArnasDickus** [zero-to-mastery/mappypals#94]
Currently, we use form.js for 7 files.
1) AddFriendForm.js
2) InvitationSent.js
3) InviteFriends.js
4) ForgotPassword.js
5) Login.js
6) ResetPassword.js
7) Signup.js

The good thing is that by reusing css we have an easier time changing the design, and making sure that design match. 
The problem: Without using !important tag you can't overwrite form.js. As a result, css has a lot of !important tags [example](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Home/AddFriendForm/AddFriendForm.css)

So the expected solution should be:
1) css should be reusable. Writing identical css for every single file would make it hard to maintain.
2) Do not use !important tags.

So what would be your solution?

**Comment by InteriorNature** [comment]
Forms that display over a map versus forms that have their own link will clear up a lot. I tested using the MapBox Container for the AddFriendForm, but no success yet. If the container is used, then problems with overriding background image and such shouldn't be a problem, and we can take out the rest of the !important uses.
- By adding closing X to Form.js, we can incorporate esc methods into other forms - I was able to remove !important when it was in Form (I also changed positioning of X to rem instead of px)
-Some things are compromises- Slightly increased button size in Form.js, so that buttons look the same across all forms (it's a compromise size). Also, altering input width to work better across all forms even when mobile.
Adding a background property in Form.js, so it always displays white, even over map helps. It was using some default before which explains why the form became transparent over the map (when the I tested removing the div wrapper css).
We probably should add a purple title option to the Form. 
We all are trying to match the mockups when we make the forms, but the variations in the mockups, lead to inconsistencies technically. We may need to make compromises on button sizes, text sizes, text arrangements etc. in order for there to be technical consistency.

**Comment by ArnasDickus** [comment]
@InteriorNature Sorry for the late response. I'm not sure I fully understand you. 
Forms that display over map [addFriendsForm](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Home/AddFriendForm/AddFriendForm.css) and [Invitation Sent](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Home/AddFriendForm/InvitationSent.css).

So you want to add closing X to form.js? I would say the problem with [form.js](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Login/Form.js) is that they select selectors instead of classes. So importing form.js and using `<button></button>` automatically stylize the button which often is not useful. Adding X class that mimics form.js I think is a good idea.

So to summarize my thoughts and your post:
1) Change form.js to select classes instead of selectors, to avoid styling issues.
2) Add X to form.js
3) Change form.js px to rem. Probably change is for the entire application.
4) Add background property of white in form.js.
5) Add a purple title in the form.js selecting by class.

So we should try to use form.js for forms as much as possible and avoid styling with our own css. Because it creates inconsistency and you can't override it without using !important?

If we agree I could create the task.

**Comment by InteriorNature** [comment]
Regarding the forms that display over maps - in mapping, everything is a layer, and so conceptually, adding a component to the top of a map involves adding it as a layer to Mapbox. I know its counterintuitive to having total control over components. But every time Mapbox re-displays, it is re-displaying in layers, it is not a single component visualization of a map. That is why there are containers for dropping a component on top, otherwise the component takes on the display logic of the map layers which distorts the sizing on forms when using Form.js.  Whether or not Form.js is used, if you want mappypals css to work only with mappypals logic, we need a container (this is how the searchbox on top is done.)

One of the reasons to use selectors in form.js actually is to help enforce uniformity across forms, and given that mappypals, at least designwise, stays within the magenta purples, it is easier to enforce uniformity. Creating a button that works across forms isn't that bad in this case. The time to start using classes, for example, is if there are multiple types of buttons needed that also have coloring differences because they have different meaning. Form.js can be upgraded so that the button selector we are using is changed to "mainbutton" class. Then, the new type of buttons become "secondarybutton" classes. So we could add that to the list you summarized. We can create  "title1", "title2", classes, and any others that have become variations.

**Comment by ArnasDickus** [comment]
So, regarding your statement about forms `But every time Mapbox re-displays, it is re-displaying in layers, it is not a single component visualization of a map. That is why there are containers for dropping a component on top, otherwise, the component takes on the display logic of the map layers which distorts the sizing on forms when using Form.js`

It seems we created forms differently. I created Addfriendform outside the interactive map. So I was able to make use of form.js without container [map.js](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Home/Map.js). While you created inside the interactive map and had to use a container in [FriendSearch](https://github.com/zero-to-mastery/mappypals/blob/dev/src/components/FriendSearch.js). 
![different-approach](https://user-images.githubusercontent.com/38657595/59961163-9bda4600-94dc-11e9-86c7-4862e456c40c.jpg) Anyways my point is that you can make use of form.js on forms that display on the map without containers. I don't see a reason why searchForm has to be inside interactive map.js

`One of the reasons to use selectors in form.js actually is to help enforce uniformity across forms, and given that mappypals, at least design-wise, stays within the magenta purples, it is easier to enforce uniformity. `

We already have multiple types of buttons:
1) X button in [AddFriendForm](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Home/AddFriendForm/AddFriendForm.css) this button needed to lose a lot of styling to make it proper.
2) Different sizes buttons: buttonReset in [ForgotPassword](https://github.com/zero-to-mastery/mappypals/blob/dev/src/pages/Login/ForgotPassword.css). We could rewrite all of it to classes instead of selectors, to avoid remove styles from X button.

There's also the idea to make a button as a component, and apply all styles to it. Example:
[button component](https://github.com/ArnasDickus/burger-app/blob/master/src/components/UI/Button/Button.jsx) 
[button css](https://github.com/ArnasDickus/burger-app/blob/master/src/components/UI/Button/Button.module.css) 
[Button in action](https://github.com/ArnasDickus/burger-app/blob/master/src/containers/Checkout/ContactData/ContactData.jsx)

Anyways, so to summarize:
1) So we agreed to Change form.js to select classes instead of selectors, to avoid styling issues.
2) Adding X button in the form.js
3) Change form.js to px to rem, because it's better for accessibility
4) Add background property of white in form.js.
5) Add a purple title in the form.js selecting by class.

So what do you think about button as a component idea? And do you agree with my summary?

**Comment by InteriorNature** [comment]
AddFriendForm is a component of the Home Page (it overlays the home page background image when it is called, and also becomes distorted in dimensions). The sizing is distorted from inserting a homepage component with the Map page inside the Map component. AddFriendForm does not work with Form.js, because of the way AddFriendForm is inserted.
Given these kinds of issues, there shouldn't be a reason for button resizing or input sizes or text sizes on forms. I agree making a button component separate makes sense - but there should be one css for forms and one css for the navigation bar (because it is not a form - it is its own functionality entity.) Otherwise, we agree on everything.
